packages/js-moi-logic/src/logic-driver.ts

```diff
@@ -175,29 +177,53 @@ export class LogicDriver<T extends Record<string, (...args: any) => any> = any>
     }
 }
 
+type LogicRoutines = Record<string, (...args: any) => any>;
+
+interface GetLogicDriver {
+    /**
+     * Returns a logic driver instance based on the given logic id.
+     * 
+     * @param {string} logicId - The logic id of the logic.
+     * @param {Signer} signer - The signer instance. 
+     * @param {Options} options - The custom tesseract options for retrieving 
+     * logic manifest. (optional)
+     * @returns {Promise<LogicDriver>} A promise that resolves to a LogicDriver instance.
+     */
+    <TLogic extends LogicRoutines>(logicId: string, signer: Signer, options?: Options): Promise<LogicDriver<TLogic>>;
+
+    /**
+     * Returns a logic driver instance based on the given logic id.
+     * 
+     * @param {string} logicId - The logic id of the logic.
+     * @param {AbstractProvider} provider - The provider instance.
+     * @param {Options} options - The custom tesseract options for retrieving 
+     * logic manifest. (optional)
+     * @returns {Promise<LogicDriver>} A promise that resolves to a LogicDriver instance.
+     */
+    <TLogic extends LogicRoutines>(logicId: string, provider: AbstractProvider, options?: Options): Promise<LogicDriver<TLogic>>;
+}
+
 /**
  * Returns a logic driver instance based on the given logic id.
  * 
  * @param {string} logicId - The logic id of the logic.
- * @param {Signer} signer - The signer or provider instance. 
- * @param {Options} options - The custom tesseract options for retrieving 
- * logic manifest. (optional)
+ * @param {Signer | AbstractProvider} value - The signer or provider instance.
+ * @param {Options} options - The custom tesseract options for retrieving
+ * 
  * @returns {Promise<LogicDriver>} A promise that resolves to a LogicDriver instance.
  */
-export const getLogicDriver = async <T extends Record<string, (...args: any) => any>>(logicId: string, signer: Signer, options?: Options): Promise<LogicDriver<T>> => {
-    try {
-        const provider = signer.getProvider();
-        const manifest = await provider.getLogicManifest(logicId, "JSON", options);
-
-        if (typeof manifest === "object") {
-            return  new LogicDriver<T>(logicId, manifest, signer);
-        }
+export const getLogicDriver: GetLogicDriver = async <T extends Record<string, (...args: any) => any>>(logicId: string, value: Signer | AbstractProvider, options?: Options): Promise<LogicDriver<T>> => {
```

Comment:
use a appropriate variable name instead of `value`.

Reply:
f251ccef800ac6a4765703c639362dd5ef8693c1